homeassistant/components/zwave_js/humidifier.py

```diff
@@ -0,0 +1,219 @@
+
+    # The setpoint type controlled by this entity
+    setpoint_type: HumidityControlSetpointType | None = None
+
+    # Suffix added to name and unique id
+    name_suffix: str | None = None
+
+
+HUMIDIFIER_ENTITY_DESCRIPTION = ZwaveHumidifierEntityDescription(
+    key="0",
+    device_class=HumidifierDeviceClass.HUMIDIFIER,
+    on_mode=HumidityControlMode.HUMIDIFY,
+    inverse_mode=HumidityControlMode.DEHUMIDIFY,
+    setpoint_type=HumidityControlSetpointType.HUMIDIFIER,
+    name_suffix="humidifier",
+)
+
+
+DEHUMIDIFIER_ENTITY_DESCRIPTION = ZwaveHumidifierEntityDescription(
+    key="1",
+    device_class=HumidifierDeviceClass.DEHUMIDIFIER,
+    on_mode=HumidityControlMode.DEHUMIDIFY,
+    inverse_mode=HumidityControlMode.HUMIDIFY,
+    setpoint_type=HumidityControlSetpointType.DEHUMIDIFIER,
+    name_suffix="dehumidifier",
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Z-Wave humidifier from config entry."""
+    client: ZwaveClient = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+
+    @callback
+    def async_add_humidifier(info: ZwaveDiscoveryInfo) -> None:
+        """Add Z-Wave Humidifier."""
+        entities: list[ZWaveBaseEntity] = []
+
+        if (
+            str(HumidityControlMode.HUMIDIFY.value)
+            in info.primary_value.metadata.states
+        ):
+            entities.append(
+                ZWaveHumidifier(
+                    config_entry, client, info, HUMIDIFIER_ENTITY_DESCRIPTION
+                )
+            )
+
+        if (
+            str(HumidityControlMode.DEHUMIDIFY.value)
+            in info.primary_value.metadata.states
+        ):
+            entities.append(
+                ZWaveHumidifier(
+                    config_entry, client, info, DEHUMIDIFIER_ENTITY_DESCRIPTION
+                )
+            )
+
+        async_add_entities(entities)
+
+    config_entry.async_on_unload(
+        async_dispatcher_connect(
+            hass,
+            f"{DOMAIN}_{config_entry.entry_id}_add_{HUMIDIFIER_DOMAIN}",
+            async_add_humidifier,
+        )
+    )
+
+
+class ZWaveHumidifier(ZWaveBaseEntity, HumidifierEntity):
+    """Representation of a Z-Wave Humidifier or Dehumidifier."""
+
+    entity_description: ZwaveHumidifierEntityDescription
+    _current_mode: ZwaveValue
+    _setpoint: ZwaveValue | None = None
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+        description: ZwaveHumidifierEntityDescription,
+    ) -> None:
+        """Initialize humidifier."""
+        super().__init__(config_entry, client, info)
+
+        self.entity_description = description
+
+        self._attr_name = f"{self._attr_name} {description.name_suffix}"
+        self._attr_unique_id = f"{self._attr_unique_id}_{description.name_suffix}"
```

Comment:
I added a `name_suffix` as part of the entity description instead of the device class